Src/FluentAssertions/Specialized/ExecutionTimeAssertions.cs

```diff
@@ -10,18 +12,41 @@ namespace FluentAssertions.Specialized
     /// </summary>
     public class ExecutionTimeAssertions
     {
-        private readonly TimeSpan executionTimeSpan;
-
-        private readonly string actionDescription;
+        private readonly ExecutionTime execution;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ExecutionTime"/> class.
         /// </summary>
         /// <param name="executionTime">The execution on which time must be asserted.</param>
         public ExecutionTimeAssertions(ExecutionTime executionTime)
         {
-            executionTimeSpan = executionTime.ExecutionTimeSpan;
-            actionDescription = executionTime.ActionDescription;
+            this.execution = executionTime;
+        }
+
+        /// <summary>
+        /// Checks the executing action if it satisfies a condition.
+        /// If the execution runs into an exeption, then this will rethrow it.
+        /// </summary>
+        /// <param name="condition">Condition to check un the current elapsed time.</param>
+        /// <param name="result">The expected result when polling is stopped.</param>
+        /// <param name="stopExecution">If polling is stopped then task execution is also stopped.</param>
+        /// <param name="rate">The rate at which the conition is re-checked.</param>
+        private void PollUntil(Func<TimeSpan, bool> condition, bool result, TimeSpan rate)
+        {
+            var forcedStop = false;
+            while (execution.IsRunning)
+            {
+                if (condition(execution.ElapsedTime) == result)
+                {
+                    break;
+                }
+                execution.Task.Wait(rate);
```

Comment:
If the return value of `Wait` is intentionally ignored, yo could use `_ = execution.Task.Wait(rate);` to explicitly state the intention.

Reply:
hah, this is a strange thing to do :)

Reply:
I like to be explicit :)
I haven't tested it out, but this should help silence static analysers reporting unused return values, when I intentionally didn't use it.